src/Controls/src/Core/Compatibility/Handlers/NavigationPage/iOS/NavigationRenderer.cs

```diff
@@ -1672,36 +1715,74 @@ void UpdateNavigationBarVisibility(bool animated)
 
 			void UpdateToolbarItems()
 			{
-				if (NavigationItem.RightBarButtonItems != null)
+				// Unsubscribe from previous toolbar item property changes
+				foreach (var item in _trackedToolbarItems)
```

Comment:
Maybe refactor this to a method called `CleanToolbarItems` , call it from here and from Disconnect method

Reply:
Good call, done!